### DISCUSSION TITLE
How can I Modify client's nickname?? Is it possible??

### DISCUSSION THREAD

**Discussion by minstudio02** [wxyz-abcd/node-haxball#42] [Q&A]
I wonder about that because I wanna modify client's name...

**Comment by wxyz-abcd** [comment]
Depends on what exactly you mean by "client's name". When you are room host, you can modify the nicknames of players that join your room on join-time. You can do this by using room.modifyPlayerDataBefore, room.modifyPlayerDataAfter or Plugin.modifyPlayerData. You can look at a file called modifyPlayerData.js at the examples folder of this repo for an example usage.

**Comment by wxyz-abcd** [comment]
You can also create a fake leave event followed by a fake join event. You are able to change not just the nickname, but everything, and it also works at any time, not just at join-time. Example usage:

```js
function changePlayerAttributes(playerId, newName, newFlag, newAvatar, newConn, newAuth){
  var oldPlayerProperties = room.fakePlayerLeave(playerId);
  // if you need, oldPlayerProperties has these attributes: { id, name, flag, avatar, conn, auth }
  room.fakePlayerJoin(playerId, newName, newFlag, newAvatar, newConn, newAuth);
}
```
